core/src/test/java/org/elasticsearch/search/sort/ScriptSortBuilderTests.java

```diff
@@ -0,0 +1,104 @@
+        }
+        if (randomBoolean()) {
+            builder.setNestedFilter(RandomSortDataGenerator.nestedFilter(builder.getNestedFilter()));
+        }
+        if (randomBoolean()) {
+            builder.setNestedPath(RandomSortDataGenerator.randomAscii(builder.getNestedPath()));
+        }
+        return builder;
+    }
+
+    @Override
+    protected ScriptSortBuilder mutate(ScriptSortBuilder original) throws IOException {
+        ScriptSortBuilder result;
+        if (randomBoolean()) {
+            // change one of the constructor args, copy the rest over
+            Script script = original.script();
+            ScriptSortType type = original.type();
+            if (randomBoolean()) {
+                result = new ScriptSortBuilder(new Script(script.getScript() + "_suffix"), type);
+            } else {
+                result = new ScriptSortBuilder(script, type.equals(ScriptSortType.NUMBER) ? ScriptSortType.STRING : ScriptSortType.NUMBER);
+            }
+            result.order(original.order());
+            result.sortMode(original.sortMode());
+            result.setNestedFilter(original.getNestedFilter());
+            result.setNestedPath(original.getNestedPath());
+            return result;
+        }
+        result = new ScriptSortBuilder(original);
+        switch (randomIntBetween(0, 3)) {
+            case 0:
+                if (original.order() == SortOrder.ASC) {
+                    result.order(SortOrder.DESC);
+                } else {
+                    result.order(SortOrder.ASC);
+                }
+                break;
+            case 1:
+                result.sortMode(RandomSortDataGenerator.mode(original.sortMode()));
+                break;
+            case 2:
+                result.setNestedFilter(RandomSortDataGenerator.nestedFilter(original.getNestedFilter()));
+                break;
+            case 3:
+                result.setNestedPath(original.getNestedPath() + "_some_suffix");
+                break;
+        }
+        return result;
+    }
+
+    public void testScriptSortType() {
+        // we rely on these ordinals in serialization, so changing them breaks bwc.
+        assertEquals(0, ScriptSortType.STRING.ordinal());
+        assertEquals(1, ScriptSortType.NUMBER.ordinal());
+
+        assertEquals("string", ScriptSortType.STRING.toString());
+        assertEquals("number", ScriptSortType.NUMBER.toString());
+
+        assertEquals(ScriptSortType.STRING, ScriptSortType.fromString("string"));
+        assertEquals(ScriptSortType.STRING, ScriptSortType.fromString("String"));
+        assertEquals(ScriptSortType.STRING, ScriptSortType.fromString("STRING"));
+        assertEquals(ScriptSortType.NUMBER, ScriptSortType.fromString("number"));
+        assertEquals(ScriptSortType.NUMBER, ScriptSortType.fromString("Number"));
+        assertEquals(ScriptSortType.NUMBER, ScriptSortType.fromString("NUMBER"));
```

Comment:
Maybe add a test or two that check the exceptions you are throwing in case of invalid values? Otherwise the string to enum part of the PR looks good.